torch/csrc/jit/codegen/cuda/ir_iostream.cpp

```diff
@@ -848,15 +864,37 @@ void IRPrinter::handle(const kir::Allocate* a) {
         os << "// Allocate global tensor ";
         break;
       case MemoryType::Shared:
-        os << "__shared__ ";
+        if (a->size()->isConstScalar()) {
+          // Static Shared Memory
+          os << "__shared__ ";
+        }
         break;
       case MemoryType::Local:
         break;
     }
-    os << a->buffer_type();
-    os << " T" << tv->name() << "[";
-    print_inline(a->size());
-    os << "];\n";
+
+    // Dynamic Shared Memory
+    if (tv->getMemoryType() == MemoryType::Shared &&
+        !a->size()->isConstScalar()) {
+      // Shared Memory Pointer
```

Comment:
for this kind of bump allocator we should be careful to make sure the pointer is naturally aligned for the data type